File: 3959-check-good-integer/3959-check-good-integer.ts
function checkGoodInteger(n: number): boolean {
    let ds = 0;
    let ss = 0;

    while (n > 0) {
        const curr = n % 10;
        ds += curr;
        ss += curr * curr;
        n = Math.floor(n / 10);
    }

    return ss - ds >= 50;
}

File: 3959-check-good-integer/3959-check-good-integer.swift
class Solution {
    func checkGoodInteger(_ n: Int) -> Bool {
        var num = n
        var ds = 0
        var ss = 0

        while num > 0 {
            let curr = num % 10
            ds += curr
            ss += curr * curr
            num /= 10
        }

        return ss - ds >= 50
    }
}